test/rules/no-unnecessary-else/test.ts.lint

```diff
@@ -0,0 +1,150 @@
+const testReturn = () => {
+    if () {
+       return;
+       ~~~~~~~  [return]
+    } else {
+       return;
+    }
+}
+
+const testReturn = () => {
+    if () {
+        if () {
+        return;
+        ~~~~~~~  [return]
+        } else {
+            return;
+        }
+    }
+}
+
+//valid
+const testReturn = () => {
+    if () {
+       return;
+    }
+    return;
+}
+
+const testReturn = () => {
+    if () {
+       return;
+       ~~~~~~~  [return]
+    } else if {
+        if () {
+            return ;
+        } else {
+            return ;
```

Comment:
Here's an example of the rule not giving a complaint when it should. Unclear if it's from the syntax errors in this test file or from the structural notes above.

Reply:
Done https://github.com/palantir/tslint/pull/4502/commits/89c814eb7648cbe716af317adb05add694305430